akka-http/server/src/main/scala/endpoints4s/akkahttp/server/Endpoints.scala

```diff
@@ -85,16 +57,21 @@ trait EndpointsWithCustomErrors
       *   - a value `Invalid(errors)` in case the URL and headers had validation
       *     errors.
       */
-    private[server] def matchAndParseHeadersDirective: Directive1[Validated[(UrlData, HeadersData)]]
+    def matchAndParseHeadersDirective: Directive1[Validated[UrlAndHeaders]]
```

Comment:
Making these public is a good idea IMO. We also had to create a fork where we just made these public to implement some custom behavior ourselves.